docs/DeveloperGuide.adoc

```diff
@@ -819,14 +819,159 @@ _{More to be added}_
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Update pet owner's contact information
+
+*MSS*
+
+1.  User requests to find a pet patient with the name "Joseph"
+2.  AddressBook shows a list of pet patients with the name "Joseph"
+3.  User requests to view contact information of a specific pet patient
+4.  AddressBook shows the pet patient's owner's contact information
+5.  User requests to modify the pet patient's owner's contact information
+6.  AddressBook shows the pet patient's updated contact information
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+* 5a. The given command is invalid.
++
+[none]
+** 5a1. AddressBook shows the correct command usage to edit pet patient's information.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Retrieve medical history of a pet patient
+
+*MSS*
+
+1.  User requests to find a pet patient with the name "Joseph"
+2.  AddressBook shows a list of pet patients with the name "Joseph"
+3.  User requests to view medical history of a specific pet patient
+4.  AddressBook shows the pet patient's medical history
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+* 4a. No medical history (empty).
++
+Use case ends.
+
+[discrete]
+=== Use case: Retrieve information of a particular owner (contact details, pet patients owned by him/her, and appointments made)
+
+*MSS*
+
+1.  User requests to find an owner named "Mavis"
+2.  AddressBook shows a list of owners with "Mavis" as part of their name
+3.  User requests to view information of a specific owner
+4.  AddressBook shows information about the owner
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Generate a pdf report of a pet patient's medical history and owner's contact information
+
+*MSS*
+
+1.  User requests to find a pet patient with the name "Joseph"
+2.  AddressBook shows a list of pet patients with the name "Joseph"
+3.  User requests for a pdf report of a specific pet patient on the list
+4.  AddressBook generates a pdf report for the user
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Generate a pdf report of a particular owner's full information e.g. contact details, list of pet patients under this owner and their medical histories
```

Comment:
Hmm did we mention "generate PDF report" during our user stories discussion? Because I don't recall any of us mentioning it; do correct me if I am wrong!

Reply:
We discussed a situation whereby vet clinic A may need to transfer a pet patient to vet clinic B due to reasons such as: owner moved house, vet clinic A unable to provide the necessary treatment for the pet patient. For transfer cases, it will be necessary to send the pet patient's medical records to vet clinic B. This is where the above use case comes it. To generate a pdf report to facilitate transfer case, or even just for reference purposes within vet clinic A.

Reply:
@Aquarinte Thanks for the clarification!

@CS2103JAN2018-F14-B2/developers Please take note of the above conversation :) This use case will come in handy if we want to continue the development of this product to v2.0.

Reply:
@Aquarinte Thanks for the clarification!

@CS2103JAN2018-F14-B2/developers Please take note of the above conversation :) This use case will come in handy if we want to continue the development of this product to v2.0.